### DISCUSSION TITLE
Z axis crashing into the bed

### DISCUSSION THREAD

**Discussion by toyprints** [Jyers/Marlin#1778] [Q&A] [answered]
Hi everyone. New E3V2 owner. Im having an issue where the print head is crashing into the end stop and just tries to keep on going. i replaced the switch and checked the wiring. I have a 4.2.2 board and no BL Touch. I've flashed the firmware several times now trying to see if that is the problem. I've bounced back and forth between here and Crealitys official site for links. I don't know what i did but nothing is working. It's like the Z limit switch doesnt tell the machine to stop when its pressed. It doesn't trigger if i manually raise the Z axis (via the menu and motor controls, not by hand) either. Is my board funny or have i screwed things up? I've been searching for over a day now and find no answers yet. currently I  have UBL - no probe - 5.5 4.2.2 installed. I have used the default 4.2.2 .bin too to no avail. 
Does anyone have any pointers?

**Comment by K2Kevin** [comment]
Check the wiring, also... manually (disable steppers in menu), move the hotend (print head) to that far left side and lower the gantry until you hear a click on the switch.  Is the HotEnd it touching your bed?  If so, raise the switch and tighten it down. IMPORTANT... be sure that switch is aligned on the rail and not tilted. 

Once the Z-limit switch height is set, time to re-level your bed.

**Reply by toyprints** [reply]
Hi and thank you for your reply. I had checked the wiring before buying a new switcfh. It's all good as far as I can see. Everything on the board is snug and nothing looks burned. The switch is moved up a bit to stop the nozzle from hitting the glass in order to troubleshoot the issue. It is at a right angle. When the rail with the print head comes down, it just keeps trying to drive through the switch. 
 I've tried to adjust the home position via the menu and it doesn't do as I assume it should.

**Reply by K2Kevin** [reply]
Then the only thing left is bad wire, wire not plugged into main board or in wrong position, or bad main board.  right?

**Reply by toyprints** [reply]
Yeah I guess so. I was hoping there was a setting I hadn't found. Im really bummed, I got a couple nice prints right out of the box. I can probably find the wires at Micro Center or Amazon, right?
 Thanks for your help!

**Reply by K2Kevin** [reply]
> Yeah I guess so. I was hoping there was a setting I hadn't found. Im really bummed, I got a couple nice prints right out of the box. I can probably find the wires at Micro Center or Amazon, right?
> Thanks for your help!

Im surprised you could ever get good prints without it being able to home properly.  hmmmm  WHAT ABOUT Z-Offset?  What is yours set to? Have you ever changed it?

**Reply by toyprints** [reply]
it was homing just fine and then all of a sudden it wasn't. I've tried adjusting the Z offset aND THAT DOESNT SEEM TO MATTER, sorry caps laptop, either. It just keeps pushing down. I just tried pulling the wires and reseating them, no help. It's almopst like the printer doesnt do whats saved. I tried to change the home position, no help either. i guess Ill have to get a new board and see if that does the trick.

**Reply by K2Kevin** [reply]
What is the current Z-offset setting?  Also, can you show a pic of the "Z-motor and rod" just behind the rail at the bottom.

**Comment by toyprints** [comment, marked as answer]
Offset is set at 0.0. 
Pic attached. 
![IMG_20220114_123822161](https://user-images.githubusercontent.com/97749756/149561306-bde1c384-b30a-4e47-9542-6127e3747b0f.jpg)

**Reply by K2Kevin** [reply]
Well...  I guess its time to install an ABL (CR-TOUCH) and ditch that Z-Stop switch altogether.  You'll thank yourself anyway.

**Reply by toyprints** [reply]
You'd go that way over a new board?

**Reply by K2Kevin** [reply]
> You'd go that way over a new board?

Well, I would contact Creality and explain so its covered under warranty, and STILL get a CR-TOUCH.

**Reply by toyprints** [reply]
Smooth. I shall do that.

**Comment by toyprints** [comment]
CR over BL?

**Reply by K2Kevin** [reply]
I like the CR-Touch.  Both do the same thing.

**Reply by toyprints** [reply]
I was kind of wondering if I had installed firmware meant to support a touch sensor, even though I didn't have a cr or bl and it just, like, shut the switch off.

**Reply by K2Kevin** [reply]
That would cause this issue for sure!

**Reply by toyprints** [reply]
I'll pick up a touch sensor. Thank you very much for all your help!!

**Reply by K2Kevin** [reply]
Be sure you are NOT using a BLtouch firmware right now.  ONLY the manual mesh leveling firmware for you right now.

**Reply by toyprints** [reply]
I pulled the file from the "release" section here for the 4.2.2 default. I checked the board to be sure it matched. No changes in the Z dive. 
 I went to creality3d and flashed the 4.2.2 stock file from there. It didn't return control to the limit switch either. 

 I formatted the sd card each time to be sure, in fat32. 

 I tested auto home, watched it hit the limit switch on the Z home and turned off the power. I left the micro SD card installed to attempt a fresh flash and I pulled it out and restarted as well. 

I opened the motherboard cover, inspected the wiring and pulled the connector and reseated it with needle nose pliers. The pins are straight. It's the third one from the bottom, small connectors. 

I tried Z offset up to +5.0 incrementally and when homing the Z axis would continue to the lower frame without stopping at the limit switch. No end stop whatsoever other than the t bolts and the switch frame. I replaced the end stop switch. 

I tried setting the home position and physically moving the stop switch to match. Stop switch is still null. The printer didn't stop homing down at the saved point anyhow. 

The switch is not active. It's brand new, creality branded. I don't think anything is wrong with the original.  

 I guess I have my filament sensor when it is running. 

I think that's everything I've done.

**Reply by K2Kevin** [reply]
When you install firmware, be sure to reset to defaults, and save again with reset defaults. Be sure that Z-stop is still set to 0.  Not 100% about the mother pin placement, but 3rd. Position from the "right side" seems correct.  Also not sure about your filament sensor, do you think it started after adding that?  Im just surprised the issue would be the main board if everything else works fine, and it worked previously

**Reply by K2Kevin** [reply]
### **Be sure that rod is TIGHT to the z-axis coupler!!! ALSO The Z-Limit Switch is tight**
Also, do you notice any binding when move the gantry down manually?  it should be very easy and smooth to move down with one finger.  **Be sure that rod is TIGHT!!!**

disable steppers, raise gantry up high while helping it by turning the z-motor coupler (or a knob you may have added to top of screw), then lightly push down.  should be very easy going down without any binding.

**Reply by toyprints** [reply]
I don't have the sensor yet. Just an extra switch right now. 

 The rail moves very smoothly. It will actually drop, slowly, on it's own from the top. It moves a little and stops. Then goes a little more. 
 
When you asked for a pic, did you want it from the power supply side? I may have misunderstood what angle you wanted. 

The only thing I haven't mentioned is adding yellow springs and the silver metal extruder. I then attempt to print. Trammed quickly although the springs were really tight. 

The rod. I have it in the coupler with roughly one or two millimeters gap. Should it seat firmly and flush inside the coupler? I watched Fat Dragons build video while putting it together. 

The issue started after a failed print after adding the springs and metal extruder. I think the room was to cold and I had very little adhesion. After I let the furnace run for a bit I tried to restart the process, homing and leveling to print. Bang bang bang into the glass. 

 I will try flashing firmware one more time with your directions. Reset to default, save, reset again. 

If that fixes it I'll let you know. I'll be picking up a cr touch within a few days. If that fixes it, I'll let you know. If it doesn't... Maybe a new board will.

**Reply by K2Kevin** [reply]
Sounds all perfect. Just make sure that rod is tight in that coupler, I would check all screws on that coupler.  BTW, When you home, the hotend homes to the edge of the glass right (not middle of the bed)?

The CR-Touch will allow you to unplug/remove that z-stop switch (No longer used). That will be very telling.  You will have to install the jyers firmware for BLtouch (I use the 5x5 mesh).

**Reply by toyprints** [reply]
The hot end will travel to the left bottom corner. The one closet to your left hand if your facing it from the front. I think that would be bottom. 

I'll double check the screws in the coupler. I'm hoping the touch probe straightens this out. It'll probably be Sunday before I can make it to microcenter.

**Reply by K2Kevin** [reply]
Interesting...  A bug in jyers/Marlin could cause this?  That would be great news!  I am using  v2.0.2h for the preview feature (BTW please add the script for download as well in your post, there are a few floating around  (That I had installed) that do not work)

**Reply by toyprints** [reply]
@tititopher68-dev hi! Thanks for the info. This issue would affect a machine without a BL Touch installed? 
 I will try your suggestion later tonight when I have time.

**Reply by K2Kevin** [reply]
The Cura_JPEG_Preview.py  that I had in scripts folder, assumed it was for this firmware.
--------------------------------------
 Cura JPEG Thumbnail creator
 Professional firmware for Ender3v2
 Miguel A. Risco-Castillo
 2021-07-01
 Contains code from:
 https://github.com/Ultimaker/Cura/blob/master/plugins/PostProcessingPlugin/scripts/CreateThumbnail.py
--------------------------------------

It did not work.  The previous plugin options/view was also different (entries for size).  Found the correct one.

**Reply by K2Kevin** [reply]
I see...  Im not up on my Marlin drama, nor do I want to be.  It's included you say?  In the pre-compiled zip?  Hmmm.  I searched and didn't have it.

**Reply by K2Kevin** [reply]
OH, I should clarify.  I did find it in the source.  NOT in the pre-compiled versions (Which is what I use)

**Reply by K2Kevin** [reply]
NOT in the pre-compiled .zip  version (Which is what I use)

**Reply by K2Kevin** [reply]
> @K2Kevin
> in this folder : `Slicer post-processing scripts for Gcode Preview`

Add that folder of files to the Pre-Compiled zip as well, and you're golden!

**Comment by K2Kevin** [comment]
Wait, did you try installing stock Creality firmware as well?  And still had your Z-crash issue.

**Reply by K2Kevin** [reply]
How many months has this buggy software been downloaded and not resolved?  How many printer affected?  How embarrassing...

**Comment by BlueWings172** [comment]
@toyprints 

See the photo below. Check the cable **from the sensor side**. I understand that you checked from motherboard side. 

check if the wires are fully inserted in the white housing of the connector that goes inside the endstop. Try to pull gently on each of the wires and see if they come out of the housing.

I remember when I got my Ender 3 V2, I had issues with the Z endstop in the first weeks. I found out that it is a common issue with the Ender 3 V2 and that in my case one of the wires became loose and wasn't making contact.

Check the wire before investing in new board. You can use the X endstop wire or sensor to test. Buying a BL touch won't be a waste of money because it is an essential upgrade and will replace your z endstop.

Just out of curiosity, aren't the new Ender 3 V2's coming with 4.2.7 board now ?


Note: don't mind the markings of cable in the photo below, it is not the original and I don't use the sensor all together.



![20220115_024136](https://user-images.githubusercontent.com/65966061/149599626-2df5e156-9799-47af-9333-5c2dd7500469.jpg)

**Reply by K2Kevin** [reply]
good call, good be that connector since two switches have failed him.  I doubt its the above software bug for this issue of z-stop crash.

**Reply by BlueWings172** [reply]
A way to test if the sensor(s) is connected to your board is to use M119 from the terminal.

You can use Octoprint terminal and run M119 and it will report something like this:

```
Recv: x_min: open
Recv: y_min: open
Recv: z_min: TRIGGERED
Recv: z_probe: TRIGGERED
Recv: filament: TRIGGERED

```


Then disable steppers and manually move print head and bed till they reach and trigger all 3 endstops then run M119 again and see if printer reports a different state for each of these endstops.

**Reply by K2Kevin** [reply]
I have never tried G-Code commands in Cura over USB,  wonder if that would work too.?.?

**Reply by BlueWings172** [reply]
@K2Kevin 

I don't think that Cura offers a terminal to communicate with the printer. If you don't have Octoprint, you may wanna use Pronterface and run it on PC.

**Reply by K2Kevin** [reply]
I have used Pronterface (still have it on Mac).  But I don't have the issues with this.  Will try when this print finishes (few minutes)

**Reply by K2Kevin** [reply]
I know cura does have a field to run commands over USB when connected, don't know if it will display any response.  Curious...  Don't feel like looking that up, Ill just try it.

**Reply by K2Kevin** [reply]
Works perfectly.  Great idea to check that switch.  Not sure if OP is still watching.

Connecting...
Printer is now online.
>>> M119
SENDING:M119
Reporting endstop status
x_min: open
y_min: open
z_min: TRIGGERED
z_probe: TRIGGERED
filament: TRIGGERED

Reporting endstop status
x_min: TRIGGERED
y_min: open
z_min: TRIGGERED
z_probe: TRIGGERED
filament: TRIGGERED

**Reply by toyprints** [reply]
I will take a better look at the connector. I did a once over when I built the printer. That wire _is_ a bit short in my opinion. An extra half inch would be great. 
 I purchased a refurbished machine. Which, can be hit or miss I know. My first two prints after building the printer were great. Then I fiddled with stuff and here we are. 
Thank you for your insight and the time you took to post a possible solution.

**Reply by K2Kevin** [reply]
Open the main board case and cut the zip tie holding that cable, you should be able to gain a little wiggle room.  Sounds like its very possible that cable connection could be an issue, like we thought. I didn't realize this was used machine, you never know where its been.

**Comment by toyprints** [comment]
Lots to catch up on here. I'll be back in the morning.

**Reply by K2Kevin** [reply]
> @toyprints 
> 
> And note : Especially not a CRTouch, It poses a lot of other problems (does not work in HS mode, loss of precision beyond 16 points, etc... a lot of feedback in this direction in many forums)... rather a BLTouch!

I haven't had any of these issues, in fact the CRTouch has performed perfectly without all the "bugs" of the BLTouch.  People have there preference, I don't feel one is any better then the other.  Now if your talking about quality, fit and finish, CRTouch wins hands down, IMO.

**Comment by toyprints** [comment]
Ok. So I switched x and z. Both fail to home. 

Installed compiled firmware and it still fails on Z. 

Id like to get a touch sensor but I'm still not sure that will be the fix. 

Contacted the dealer. They're most likely going to ship a new board. 

Thanks for all the help.

**Reply by K2Kevin** [reply]
@tititopher68-dev  Thanks for the info. With non HS firmware, is there still an issue to watch for?

**Comment by toyprints** [comment]
Quick update. Decided to gamble on the BL Touch. 
![16430970204841491404433938222410](https://user-images.githubusercontent.com/97749756/150934183-6d563ce3-97c2-460e-a5a1-95601e819663.jpg)
A little hairspray and we're doing it. A little stringy. It's also cold in here. 
![16430971115516871775892426897436](https://user-images.githubusercontent.com/97749756/150934443-e9bdbf99-3651-4362-b720-2d2d4d2ec9ea.jpg)
Bed appears to be about as level as I can get. 

On to learning Cura. 
Thank you for all the help, again. Time to lay down some dungeon tiles!

**Comment by BlueWings172** [comment]
@toyprints 

Looks good! Hope this ended your issues. You may wanna give UBL a try.

Curious was kind of filament you are printing at 210/85 C.

**Reply by toyprints** [reply]
I'm using grey inland tough pla. I turned the extruder up to 221 and the bed down to 75. After a soap and water cleaning of the bed, the 1st layer was near perfect on my first dungeon tile. No hairspray needed at all. 
What is UBL?

**Reply by toyprints** [reply]
Oh wait, is that the mesh bed leveling in Marlin? I think I've got that down pat. I leveled the bed by eye while printing the stripes and then dialed in by stepping the z axis down/up. I e added the G29 command to my Cura start commands. It does appear to measure the bed before the prints, but I'm not positive it's working properly. It stops at 9 points and then begins printing. How do I know if it saved that mesh though?Still having some adhesion issues on the carborundum glass bed. However, I did not clean it between prints.
 Does anyone use the silicone bed sleeves instead of the springs? I'm wondering if they'll help keep it in alignment.

**Reply by BlueWings172** [reply]
@toyprints 

If the printer probes 9 points and stops, then you are using 3x3 mesh bed leveling. If you wanna be more accurate, you can use 5x5 which probes 25 points (you need to use the 5x5 points firmware for that). So more points, more accuracy. That said, it becomes time consuming if you are going to probe 25 points before every print, especially when you are doing calibration prints that take just 10 minutes.

So here comes UBL, which is the next level auto bed leveling solution. I use 15x15 UBL which probes around 155 points then saves it to the machine.  You only have to do this once every week or month since the imperfections in your bed surface do not change.

Then before each print, only 3 points are probed and the machine start printing quickly. So with UBL you get more accuracy + much less probing time before each print.

To use UBL, you need to use necessary firmware, then add the below to your Cura starting code:

G29 L0 ; Load default mesh from EEPROM.
G29 A ; Activate UBL.
G29 J1 ; Get the bed tilt in 3-points.

don't forget, first time you have to go the Level Menu then Create New Mesh

Takes about 20 minutes then saves to memory.
